pkg/services/ngalert/schedule/schedule.go

```diff
@@ -311,7 +322,7 @@ func (sch *schedule) processTick(ctx context.Context, dispatcherGroup *errgroup.
 		toDelete = append(toDelete, key)
 	}
 	sch.DeleteAlertRule(toDelete...)
-	return readyToRun, registeredDefinitions
+	return readyToRun, registeredDefinitions, pausedAlertRuleKeys
```

Comment:
What I don't understand is no code uses `pausedAlertRuleKeys`? Is it just for tests?

Reply:
Yes, it is only for tests. Same as `readyToRun` and `registeredDefinitions` objects.

Reply:
Yes, this is done for tests as it is the only straightforward way to test the logic that decides what is supposed to be run\deleted\paused at the current tick. Feel free to refactor it.